### PR TITLE
feat: updates and enhancements

### PR DESCRIPTION
**Pull request by nasbench**

### CODE REVIEW

File: rules/windows/process_creation/proc_creation_win_fltmc_unload_driver_sysmon.yml [hunk 1]
```diff
@@ -1,12 +1,12 @@
-title: Sysmon Driver Unload
+title: Sysmon Filter Driver Unloaded Via Fltmc.EXE
 id: 4d7cda18-1b12-4e52-b45c-d28653210df8
```
Comment (frack113) [inline rules/windows/process_creation/proc_creation_win_fltmc_unload_driver_sysmon.yml:2]:
```suggestion
id: 4d7cda18-1b12-4e52-b45c-d28653210df8
related:
    - id: 4931188c-178e-4ee7-a348-39e8a7a56821 # Generic
      type: similar
```
Will get the 2 alerts

File: rules/windows/process_creation/proc_creation_win_lolbin_squirrel.yml [hunk 2]
```diff
@@ -2,30 +2,94 @@ title: Use of Squirrel.exe
...
+    filter_teams:
+        CommandLine|contains|all:
+            - 'C:\Users\'
+            - '\AppData\Local\Microsoft\Teams\Update.exe'
+            - 'Teams.exe'
+        CommandLine|contains:
+            - '--processStart'
+            - '--createShortcut'
+    condition: selection_img and (all of selection_download_* or selection_exec) and not 1 of filter_*
 falsepositives:
```
Comment (frack113) [inline rules/windows/process_creation/proc_creation_win_lolbin_squirrel.yml:62]: I prefer a more generic phrase than a list

File: rules/windows/process_creation/proc_creation_win_pua_runxcmd.yml [hunk 3]
```diff
@@ -15,11 +16,13 @@ logsource:
     category: process_creation
     product: windows
 detection:
-    selection:
+    selection_account:
         CommandLine|contains|all:
```
Comment (frack113) [inline rules/windows/process_creation/proc_creation_win_pua_runxcmd.yml:20]:
```suggestion
        CommandLine|contains:
```

**Review by frack113 — APPROVED** [review]

File: rules/windows/process_creation/proc_creation_win_lolbin_squirrel.yml [hunk 4]
```diff
@@ -2,30 +2,63 @@ title: Use of Squirrel.exe
 id: 45239e6a-b035-4aaf-b339-8ad379fcb67e
 related:
     - id: fa4b21c9-0057-4493-b289-2556416ae4d7
-      type: derived
+      type: obsoletes
 status: experimental
-description: The "Squirrel.exe" binary that is part of multiple software (Slack, Teams, Discord...etc) can be used as a LOLBIN
+description: Detects usage of the "Squirrel.exe" binary as a LOLBIN. This binary is part of multiple software installations (Slack, Teams, Discord...etc)
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_lolbin_squirrel.yml:7]:
```suggestion
description: Detects the usage of the "Squirrel.exe" binary as a LOLBIN. This binary is part of multiple software installations (Slack, Teams, Discord, etc.)
```

File: rules/windows/process_creation/proc_creation_win_malware_conti_ransomware_commands.yml [hunk 5]
```diff
@@ -1,13 +1,13 @@
-title: Conti Ransomware Execution
+title: Potential Conti Ransomware Activity
 id: 689308fc-cfba-4f72-9897-796c1dc61487
 status: test
-description: Conti ransomware command line ioc
+description: Detects specific command used by the Conti ransomware group
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_malware_conti_ransomware_commands.yml:4]:
```suggestion
description: Detects a specific command used by the Conti ransomware group
```

File: rules/windows/process_creation/proc_creation_win_netsh_fw_add_rule.yml [hunk 6]
```diff
@@ -1,26 +1,27 @@
-title: Netsh Port or Application Allowed
+title: New Firewall Rule Added Via Netsh.EXE
 id: cd5cfd80-aa5f-44c0-9c20-108c4ae12e3c
 status: test
-description: Allow Incoming Connections by Port or Application on Windows Firewall
+description: Detects the addition of a new rule to the windows firewall via netsh
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_netsh_fw_add_rule.yml:4]:
```suggestion
description: Detects the addition of a new rule to the Windows firewall via netsh
```

File: rules/windows/process_creation/proc_creation_win_netsh_fw_allow_program_in_susp_location.yml [hunk 7]
```diff
@@ -1,61 +1,60 @@
-title: Netsh Program Allowed with Suspcious Location
+title: Suspcious Program Allowed Through The Firewall Via Netsh.EXE
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_netsh_fw_allow_program_in_susp_location.yml:1]:
```suggestion
title: Suspicious Program Location Whitelisted In Firewall Via Netsh.EXE
```

File: rules/windows/process_creation/proc_creation_win_netsh_fw_allow_program_in_susp_location.yml [hunk 8]
```diff
@@ -1,61 +1,60 @@
...
 id: a35f5a72-f347-4e36-8895-9869b0d5fc6d
 status: test
-description: Detects Netsh commands that allows a suspcious application location on Windows Firewall
+description: Detects Netsh command execution that allows an application located in a suspcious location through the Windows Firewall
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_netsh_fw_allow_program_in_susp_location.yml:4]:
```suggestion
description: Detects Netsh command execution that whitelists a program located in a suspicious location in the Windows Firewall
```

File: rules/windows/process_creation/proc_creation_win_netsh_fw_allow_rdp.yml [hunk 9]
```diff
@@ -0,0 +1,38 @@
+title: RDP Connection Allowed Via Netsh.EXE
+id: 01aeb693-138d-49d2-9403-c4f52d7d3d62
+status: test
+description: Detects usage of the netsh command to opens and allow connections to port 3389 (RDP). As seen used by Sarwent Malware
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_netsh_fw_allow_rdp.yml:4]:
```suggestion
description: Detects usage of the netsh command to open and allow connections to port 3389 (RDP). As seen used by Sarwent Malware
```

File: rules/windows/process_creation/proc_creation_win_netsh_helper_dll_persistence.yml [hunk 10]
```diff
@@ -1,15 +1,16 @@
-title: Suspicious Netsh DLL Persistence
+title: Potential Persistence Via Netsh Helper DLL
 id: 56321594-9087-49d9-bf10-524fe8479452
 status: test
-description: Detects persitence via netsh helper
+description: Detects execution of netsh with "add helper" flag in order to add a csutom helper DLLs. This technique can be abused to add a malicious helper DLL that can be used a persistence proxy that gets called when netsh.exe is executed.
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_netsh_helper_dll_persistence.yml:4]:
```suggestion
description: Detects the execution of netsh with "add helper" flag in order to add a custom helper DLL. This technique can be abused to add a malicious helper DLL that can be used as a persistence proxy that gets called when netsh.exe is executed.
```

File: rules/windows/process_creation/proc_creation_win_netsh_port_forwarding_3389.yml [hunk 11]
```diff
@@ -1,12 +1,12 @@
-title: Netsh RDP Port Forwarding
+title: RDP Port Forwarding Rule Added Via Netsh.EXE
 id: 782d6f3e-4c5d-4b8c-92a3-1d05fed72e63
 status: test
-description: Detects netsh commands that configure a port forwarding of port 3389 used for RDP
+description: Detects execution of netsh to configure a port forwarding of port 3389 (RDP) rule
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_netsh_port_forwarding_3389.yml:4]:
```suggestion
description: Detects the execution of netsh to configure a port forwarding of port 3389 (RDP) rule
```

File: rules/windows/process_creation/proc_creation_win_setspn_spn_enumeration.yml [hunk 12]
```diff
@@ -0,0 +1,29 @@
+title: Potential SPN Enumeration Via Setspn.EXE
+id: 1eeed653-dbc8-4187-ad0c-eeebb20e6599
+status: test
+description: Detects Service Principal Name Enumeration used for Kerberoasting
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_setspn_spn_enumeration.yml:4]:
```suggestion
description: Detects service principal name (SPN) enumeration used for Kerberoasting
```

File: rules/windows/process_creation/proc_creation_win_taskkill_sep.yml [hunk 13]
```diff
@@ -2,14 +2,14 @@ title: Taskkill Symantec Endpoint Protection
 id: 4a6713f6-3331-11ed-a261-0242ac120002
 status: experimental
 description: |
-  Detects one of the possible scenarios for disabling symantec endpoint protection.
-  Symantec Endpoint Protection antivirus software services incorrectly implement the protected service mechanism.
-  As a result, the NT AUTHORITY/SYSTEM user can execute the taskkill /im command several times ccSvcHst.exe /f, thereby killing the process belonging to the service, and thus shutting down the service.
+    Detects one of the possible scenarios for disabling symantec endpoint protection.
+    Symantec Endpoint Protection antivirus software services incorrectly implement the protected service mechanism.
+    As a result, the NT AUTHORITY/SYSTEM user can execute the taskkill /im command several times ccSvcHst.exe /f, thereby killing the process belonging to the service, and thus shutting down the service.
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_taskkill_sep.yml:7]:
```suggestion
    Detects one of the possible scenarios for disabling Symantec Endpoint Protection.
    Symantec Endpoint Protection antivirus software services incorrectly implement the protected service mechanism.
    As a result, the NT AUTHORITY/SYSTEM user can execute the taskkill /im command several times ccSvcHst.exe /f, thereby killing the process belonging to the service, and thus shutting down the service.
```

**Review by phantinuss — APPROVED** [review]

File: rules/windows/process_creation/proc_creation_win_fltmc_unload_driver_sysmon.yml [hunk 14]
```diff
@@ -1,12 +1,15 @@
-title: Sysmon Driver Unload
+title: Sysmon Filter Driver Unloaded Via Fltmc.EXE
```
Comment (nasbench) [inline rules/windows/process_creation/proc_creation_win_fltmc_unload_driver_sysmon.yml:1]:
```suggestion
title: Sysmon Driver Unloaded Via Fltmc.EXE
```

File: rules/windows/process_creation/proc_creation_win_fltmc_unload_driver_sysmon.yml [hunk 15]
```diff
@@ -1,12 +1,15 @@
...
 id: 4d7cda18-1b12-4e52-b45c-d28653210df8
+related:
+    - id: 4931188c-178e-4ee7-a348-39e8a7a56821 # Generic
+      type: similar
 status: test
-description: Detect possible Sysmon driver unload
+description: Detect possible Sysmon filter driver unload via fltmc.exe
```
[the remark this answers is not in the document]
Comment (nasbench) [reply rules/windows/process_creation/proc_creation_win_fltmc_unload_driver_sysmon.yml:7]: The correct term is a filter driver but I see your point. I will change the title as well.

**Pull request merged**